app/Cli.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE ApplicativeDo #-}
+
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+CLI options for the @extensions@ exectuable.
+-}
+
+module Cli
+    ( ExtensionsArgs (..)
+    , runExtensionsCli
+    ) where
+
+import Options.Applicative (Parser, ParserInfo, execParser, fullDesc, help, helper, info, long,
+                            metavar, optional, progDesc, strOption, switch)
+
+
+-- | Options used for the main @extensions@ executable.
+data ExtensionsArgs = ExtensionsArgs
+    { extensionsArgsModuleFilePath :: !(Maybe FilePath)  -- ^ Path to module
+    , extensionsArgsCabalFilePath  :: !(Maybe FilePath)  -- ^ Path to cabal file
+    , extensionsArgsNoCabal        :: !Bool  -- ^ Do not fail if no @.cabal@ file
+    , extensionsArgsNoModules      :: !Bool  -- ^ Do not print per-module info
+    }
+
+-- | Run main parser of the @extensions@ command line tool.
+runExtensionsCli :: IO ExtensionsArgs
+runExtensionsCli = execParser extensionsCliParser
+
+extensionsCliParser :: ParserInfo ExtensionsArgs
+extensionsCliParser = info (helper <*> extensionsP) $
+    fullDesc <> progDesc "Display Haskell Language Extensions"
+
+-- | @extensions@ command parser.
+extensionsP :: Parser ExtensionsArgs
+extensionsP = do
+    extensionsArgsModuleFilePath <- moduleFilePathP
+    extensionsArgsCabalFilePath  <- cabalFilePathP
+    extensionsArgsNoCabal        <- noCabalP
+    extensionsArgsNoModules      <- noModulesP
+    pure ExtensionsArgs{..}
+
+moduleFilePathP :: Parser (Maybe FilePath)
+moduleFilePathP = optional $ strOption $ mconcat
+    [ long "module-file-path"
+    , metavar "FILE_PATH"
+    , help "Relative path to the Haskell module"
+    ]
+
+cabalFilePathP :: Parser (Maybe FilePath)
+cabalFilePathP = optional $ strOption $ mconcat
+    [ long "cabal-file-path"
+    , metavar "FILE_PATH"
+    , help "Relative path to the .cabal file"
+    ]
+
+noCabalP :: Parser Bool
+noCabalP = switch $ mconcat
```

Comment:
@vrom911 Excellent idea! It will be much clearer with a custom enum. A lot of developers actually advocate for using custom enum types as much as possible instead of `Bool`.

What do you think about the following type?

```haskell
data Toggle = Disable | Enable
```